README.md

```diff
@@ -0,0 +1,269 @@
+
+### Repositories
+
+- **BookRepository**: Interface for CRUD operations on Book entities and special methods for finding books by category.
+- **ShoppingCartRepository**: Interface for CRUD operations on ShoppingCart entities and methods for identifying a shopping cart by user ID.
+- **CartItemRepository**: Interface for CRUD operations on CartItem entities.
+- **OrderRepository**: Interface for CRUD operations on Order entities.
+- **OrderItemRepository**: Interface for CRUD operations on OrderItem entities and methods for retrieving order items by order ID.
+- **UserRepository**: Interface for CRUD operations on User entities and methods for checking if the user is already registered.
+- **RoleRepository**: Interface for CRUD operations on Role entities, including methods for identifying user roles.
+
+## Setup and Usage
+
+To get started with this project, follow these steps:
+
+1. **Clone the repository**:
+    ```sh
+    git clone https://github.com/bodya4243/online-book-store.git
+    cd online-book-store
+    ```
+
+2. **Build the project using Maven**:
+    ```sh
+    mvn clean install
+    ```
+
+3. **Run the application**:
+    ```sh
+    mvn spring-boot:run
+    ```
+   Technologies and Tools
+   Key Technologies
+   Java Development Kit (JDK) 17
+   Maven
+   MySQL
+   Spring Boot
+   H2 Database (for testing)
+   Lombok
+   MapStruct
+   Liquibase
+   SpringDoc OpenAPI
+   JWT (JSON Web Token)
+   Docker Compose
+   
+
+
+4. **Access the application**:
+    - The application runs on `http://localhost:8080`.
+    - API documentation is available at `http://localhost:8080/swagger-ui.html`.
+
+
+
+### Docker Setup
+
+**Swagger**:
+   - You can use Swagger for manipulating API endpoints.
+
+**Docker**:
+   - You can download the container from Docker and start the API.
+
+*** docker-compose
+
+services:
+mysqldb:
+restart: always
+env_file: ./.env
+image: mysql:8.0
+environment:
+MYSQL_ROOT_PASSWORD: ${DB_PASSWORD}
+MYSQL_DATABASE: ${MYSQL_DATABASE}
+MYSQL_USER: ${MYSQL_USER}
+MYSQL_PASSWORD: ${MYSQL_PASSWORD}
+ports:
+- "3309:3306"
+volumes:
+- db_data:/var/lib/mysql
+healthcheck:
+test: [ "CMD-SHELL", "mysqladmin ping -h localhost -P ${DB_DOCKER_PORT} -u ${DB_USER} -p${DB_PASSWORD}" ]
+interval: 30s
+timeout: 10s
+retries: 3
+start_period: 60s
+
+app:
+depends_on:
+mysqldb:
+condition: service_healthy
+image: task-manager-api
+build: .
+env_file: ./.env
+ports:
+- "${SPRING_LOCAL_PORT}:${SPRING_DOCKER_PORT}"
+- "${DEBUG_PORT}:${DEBUG_PORT}"
+environment:
+SPRING_APPLICATION_JSON: '{
+"spring.datasource.url": "jdbc:mysql://mysqldb:${DB_DOCKER_PORT}/${DB_DATABASE}?serverTimezone=UTC",
+"spring.datasource.username": "${DB_USER}",
+"spring.datasource.password": "${DB_PASSWORD}",
+"jwt.secret": "${JWT_SECRET}"
+}'
+JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5000"
+
+volumes:
+db_data:
```

Comment:
all of that is not readable, and you do not need to have it here, this file is ready for use and does not need any clarifications on how to use it